06_predict_survival/run_survival_prediction.py

```diff
@@ -0,0 +1,246 @@
+"""
+Script to run pan-cancer mutation classification experiments, with stratified
+train and test sets, for all provided genes.
+"""
+import sys
+import argparse
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+from tqdm import tqdm
+from sksurv.exceptions import NoComparablePairException
+
+import mpmp.config as cfg
+from mpmp.data_models.tcga_data_model import TCGADataModel
+from mpmp.exceptions import (
+    ResultsFileExistsError,
+    NoTrainSamplesError,
+    NoTestSamplesError,
+    OneClassError,
+)
+from mpmp.prediction.cross_validation import run_cv_stratified
+import mpmp.utilities.data_utilities as du
+import mpmp.utilities.file_utilities as fu
+from mpmp.utilities.tcga_utilities import (
+    get_all_data_types,
+    check_all_data_types,
+)
+
+def process_args():
+    """Parse and format command line arguments."""
+
+    parser = argparse.ArgumentParser()
+
+    # argument group for parameters related to input/output
+    # (e.g. filenames, logging/verbosity options, target genes)
+    #
+    # these don't affect the model output, and thus don't need to be saved
+    # with the results of the experiment
+    io = parser.add_argument_group('io',
+                                   'arguments related to script input/output, '
+                                   'note these will *not* be saved in metadata ')
+    io.add_argument('--cancer_types', nargs='*', default=None,
```

Comment:
It's a little bit unintuitive that None = all. I guess it makes sense to do this way because there might be a string collision between a cancer type called 'ALL' and the desired function of running all cancer types

Reply:
Yeah, I guess 'ALL' would be a bit ambiguous (although there isn't any ALL data in TCGA unfortunately) but another string like `'all_cancer_types'` might be clearer.